workflows/csharp/sdk/order-processor/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Dapr.Client;
+using Dapr.Workflow;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using WorkflowConsoleApp.Activities;
+using WorkflowConsoleApp.Models;
+using WorkflowConsoleApp.Workflows;
+
+const string storeName = "statestore";
+
+// The workflow host is a background service that connects to the sidecar over gRPC
+var builder = Host.CreateDefaultBuilder(args).ConfigureServices(services =>
+{
+    services.AddDaprWorkflow(options =>
+    {
+        // Note that it's also possible to register a lambda function as the workflow
+        // or activity implementation instead of a class.
+        options.RegisterWorkflow<OrderProcessingWorkflow>();
+
+        // These are the activities that get invoked by the workflow(s).
+        options.RegisterActivity<NotifyActivity>();
+        options.RegisterActivity<ReserveInventoryActivity>();
+        options.RegisterActivity<ProcessPaymentActivity>();
+        options.RegisterActivity<UpdateInventoryActivity>();
+    });
+});
+
+// Start the app - this is the point where we connect to the Dapr sidecar
+using var host = builder.Build();
+host.Start();
+
+using var daprClient = new DaprClientBuilder().Build();
+
+// Wait for the sidecar to become available
```

Comment:
Do we really need this sidecar available code?

Reply:
The "Models.cs" file is used for grouping together common structures used across multiple files. While it isn't technically 'needed' it makes the code a lot cleaner and more organized so we should keep it in.

Waiting for the sidecar to become available is some defensive programming and leads to less errors in our automated testing/CI.